web/profiles/custom/os2loop/modules/os2loop_analytics/src/Helper/AnalyticsHelper.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Drupal\os2loop_analytics\Helper;
+
+use Drupal\os2loop_analytics\Form\SettingsForm;
+use Drupal\os2loop_settings\Settings;
+use Drupal\Core\Render\Markup;
+
+/**
+ * Helper for os2loop_analytics.
+ */
+class AnalyticsHelper {
+  /**
+   * The config.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  private $config;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\os2loop_settings\Settings $settings
+   *   Settings.
+   */
+  public function __construct(Settings $settings) {
+    $this->config = $settings->getConfig(SettingsForm::SETTINGS_NAME);
+  }
+
+  /**
+   * Set analytics tools script in head.
+   *
+   * @param array $attachments
+   *   The attachments to add.
+   */
+  public function setAnalytics(array &$attachments) {
+    $scripts = $this->getAnalyticsConfig();
+    foreach ($scripts as $name => $script) {
+      $cleanScript = $this->cleanup($script);
+      $attachments['#attached']['html_head'][] = [
+        [
+          '#tag' => 'script',
+          '#value' => Markup::create($cleanScript),
+          '#weight' => 1,
+        ],
+        $name . '_script',
+      ];
+    }
+  }
+
+  /**
+   * Get config related to os2loop analytics.
+   *
+   * @return array
+   *   A list of analytics scripts.
+   */
+  private function getAnalyticsConfig(): array {
+    $scripts = [];
+    $scripts += $this->config->get('matomo_tracking_code') ? ['matomo' => $this->config->get('matomo_tracking_code')] : [];
+    $scripts += $this->config->get('google_analytics_tracking_code') ? ['google_analytics' => $this->config->get('google_analytics_tracking_code')] : [];
+    $scripts += $this->config->get('siteimprove_tracking_code') ? ['siteimprove' => $this->config->get('siteimprove_tracking_code')] : [];
```

Comment:
I think this code will be easier to read (for a human) if written as

```php
if ($this->config->get('matomo_tracking_code')) {
  $scripts['matomo'] = $this->config->get('matomo_tracking_code');
}
if ($this->config->get('google_analytics_tracking_code')) {
  …
}
…
```

You could also go for 
```php
$scripts = [
  'matomo' => $this->config->get('matomo_tracking_code'),
  'google_analytics' => $this->config->get('google_analytics_tracking_code'),
  …
];

// Remove empty values.
return array_filter($scripts);
```

Reply:
Removed helper all together